ado-build.yml

```diff
@@ -32,3 +27,23 @@ jobs:
       script: 'BuildAndTest.cmd'
 
   - task: ComponentGovernanceComponentDetection@0
+    inputs:
+      ignoreDirectories: '..\src\sarif-sdk'
+
+- job: mac
+  pool:
+    vmImage: 'macOS-latest'
+  steps:
+  - checkout: self
+    submodules: true
+
+  - task: Bash@3
+    displayName: 'Build and Test'
+    inputs:
+      filePath: 'BuildAndTest.sh'
+
+  - task: Bash@3
+    displayName: 'Run BinSkim'
+    inputs:
+      targetType: 'inline'
+      script: 'dotnet bld/bin/x64_Release/netcoreapp3.1/binskim.dll analyze src/Test.FunctionalTests.BinSkim.Driver/BaselineTestData/macho.*'
```

Comment:
>dotnet bld/bin/x64_Release/netcoreapp3.1/binskim.dll [](http://example.com/codeflow?start=15&length=52)

we should add this in some doc.
the user would need to execute:
`dotnet binskim.dll analyze ....`

Reply:
+1

I see one candidate place,
at the bottom of the readme.md
we currently have:

`Example: binskim.exe analyze c:\bld\*.dll --recurse --output MyRun.sarif`

I think we can change it to 3 OS version example here.

we can also consider if want to include the path in the example to make it more clear, because the previous step is unzip.

Reply:
more info fyi
I found that we have this file "BinSkim" in the package
`./BinSkim analyze`
I just tried works in Linux.

Reply:
@shaopeng-gh, could you clarify where you found this file?

Reply:
download package from
[https://www.nuget.org/packages/Microsoft.CodeAnalysis.BinSkim/](https://www.nuget.org/packages/Microsoft.CodeAnalysis.BinSkim/)
(which is the result of our build and package)
unzip
go to folder \tools\netcoreapp3.1\linux-x64
file "BinSkim" without ext

Reply:
Apologies. I was confused since this change is specific to osx. You were being helpful, I was concerned you were telling me we had a binary for linux in the wrong location.

Reply:
macOS also have a same file in same location, after your change:
\tools\netcoreapp3.1\osx-x64\BinSkim

I point out in the case we want to add doc for user
see if we want to say use 
`./BinSkim analyze`
or 
`dotnet binskim.dll analyze `

Reply:
When I created this, I didn't want to change the current directory that we are right now. So, it is why we have the full path.

But, in the end, it is the same as dotnet binskim.dll analyze.

Reply:
Mary also updated the readme showing where the files will be. so, I think, Shaopeng's suggestion is to add the dotnet binskim.dll analyze as a new step in the readme

Reply:
I think, Shaopeng's suggestion is to add the dotnet binskim.dll analyze as a new step in the readme  
------ yes, I think it is not necessary, it should be clear enough, just throwing ideals, let me close this comment.